proj/src/main/webapp/script.js

```diff
@@ -69,4 +72,33 @@ function pushChatMessage() {
     const message = document.getElementById('message-input').value;
     // push message to datastore
     messageRef.push(message);
-} 
+}
+
+function addMoreMessagesAtTheTop() {
+    const dbRefObject = firebase.database().ref(path);
+    const chat = document.getElementById('chat-as-list');
+    if (chat.scrollTop === 0) {
```

Comment:
Have you considered loading when they get close to the top but before they actually reach it?  I don't know if that's the right thing to do, but it's worth considering.

Reply:
I tried do that, but it ended up triggering the function more than once even when the tolerance was really small, so I just made it so it has to be 0

Reply:
There are clever ways to deal with that, but I agree that's not a priority.